### PR TITLE
Tests implementation

### PR DESCRIPTION
**Pull request by aniaw**
Some helping functions.

### CODE REVIEW

File: test/Querier.test.ts [hunk 1]
```diff
@@ -14,25 +14,30 @@ const successQuery = async () => {
...
+  id: string = queryKey,
+  result: { name: string } = queryResult,
+  $reason: string = queryReason
+): QuerierStoreType => {
+  return { key: { id, result, state, $props, $reason } };
 };
+const storeMock: QuerierStoreType = createStore({ state: QuerierState.Active }, null, null);
 
-describe('Querier', () => {
+fdescribe('Querier', () => {
```
Comment (dprokop) [inline test/Querier.test.ts]: Typo @aniaw?

File: test/Querier.test.ts [hunk 2]
```diff
@@ -133,38 +135,34 @@ describe('Querier', () => {
 
     it('allows subscription cancelation', () => {
       const querier = new Querier();
-      const unsubscribe = querier.subscribe('key', jest.fn());
+      const subscribe = querier.subscribe('key', jest.fn());
```
Comment (dprokop) [inline test/Querier.test.ts]: `querier.subscribe` returns function that unsubscribes from changes, that's why the verbose name here

File: test/Querier.test.ts [hunk 3]
```diff
@@ -133,38 +135,34 @@ describe('Querier', () => {
...
+      let listenerSpy;
+      let spiedQuery;
+
+      beforeEach(() => {
+        querySpy = jest.fn();
+        listenerSpy = jest.fn();
+        spiedQuery = async () => {
           querySpy();
-          await successQuery();
+          return await successQuery();
```
Comment (dprokop) [inline test/Querier.test.ts]: 👍

File: test/Querier.test.ts [hunk 4]
```diff
@@ -14,25 +14,30 @@ const successQuery = async () => {
...
-      state: QuerierState.Active
-    },
-    result: null,
-    $reason: 'SomeComponent',
-    $props: {}
-  }
+
+const queryKey = 'key';
+const queryResult = users[1];
+const queryReason = 'Some Component';
```
Comment (dprokop) [inline test/Querier.test.ts]: Let's change the value to `SomeComponent` so that it's not misleading.

File: test/Querier.test.ts [hunk 5]
```diff
@@ -14,25 +14,30 @@ const successQuery = async () => {
...
 
-describe('Querier', () => {
+fdescribe('Querier', () => {
   describe('utilities', () => {
     test('getEntry', () => {
       const querier = new Querier(storeMock);
 
       expect(Object.keys(querier.getStore())).toHaveLength(1);
-      expect(querier.getEntry('queryKey')).toEqual(storeMock.queryKey);
+      expect(querier.getEntry('key')).toEqual(storeMock.key);
```
Comment (dprokop) [inline test/Querier.test.ts]: What about using `queryKey` const here? To be even more verbose, I suggest doing:

```
expect(querier.getEntry(queryKey)).toEqual(storeMock[queryKey]);
```

File: test/Querier.test.ts [hunk 6]
```diff
@@ -14,25 +14,30 @@ const successQuery = async () => {
...
-    result: null,
-    $reason: 'SomeComponent',
-    $props: {}
-  }
+
+const queryKey = 'key';
+const queryResult = users[1];
+const queryReason = 'Some Component';
+
+const createStore = (
```
Comment (dprokop) [inline test/Querier.test.ts]: I would say it's `mockStore`

File: test/Querier.test.ts [hunk 7]
```diff
@@ -177,32 +175,11 @@ describe('Querier', () => {
...
 
         expect(querySpy).not.toBeCalled();
-        expect(listenerSpy).toBeCalledWith({
-          id: queryKey,
-          result: queryResult,
-          state: { state: QuerierState.Success },
-          $props: null,
-          $reason: 'SomeComponent'
-        });
+        expect(listenerSpy).toBeCalledWith(store.key);
```
Comment (dprokop) [inline test/Querier.test.ts]: Same as above, for verbosity `store[queryKey]`

**Review by dprokop — CHANGES_REQUESTED** [review]

File: test/Querier.test.ts (hunk 1 above)
Comment (aniaw) [reply test/Querier.test.ts]: I use `fdescribe` to run one selected describe, but yes - this shouldn't be commited.

File: test/Querier.test.ts (hunk 2 above)
Comment (aniaw) [reply test/Querier.test.ts]: Ok!

File: test/Querier.test.ts (hunk 1 above)
Comment (dprokop) [reply test/Querier.test.ts]: Nice, didn't know about that :)

File: test/Querier.test.ts [hunk 8]
```diff
@@ -29,15 +29,15 @@ const createStore = (
 ): QuerierStoreType => {
   return { key: { id, result, state, $props, $reason } };
 };
-const storeMock: QuerierStoreType = createStore({ state: QuerierState.Active }, null, null);
+const mockStore: QuerierStoreType = createMockStore({ state: QuerierState.Active }, null, null);
```
Comment (dprokop) [inline test/Querier.test.ts]: Haha, I guess we misunderstood each other here ;) I meant to change `createStore` to `createStoreMock`. `storeMock` const name was perfectly fine.

**Pull request closed without merging**